src/service/viewer-impl.js

```diff
@@ -68,19 +68,34 @@ export const ViewportType = {
 /**
  * Visibility state of the AMP document.
  * @enum {string}
- * @private
  */
 export const VisibilityState = {
-
   /**
-   * Viewer has shown the AMP document.
+   * The AMP document is currently active and visible.
    */
   VISIBLE: 'visible',
 
   /**
-   * Viewer has indicated that AMP document is hidden.
+   * The AMP document is active but the browser tab is not.
+   */
+  HIDDEN: 'hidden',
+
+  /**
+   * The AMP document is visible but it should stop active playback because
+   * the user is swiping the doc.
+   */
+  PAUSED: 'paused',
+
+  /**
+   * The AMP document is no longer active because the user swiped away or
```

Comment:
Just confirmed that the viewer does send the `HIDDEN` state (which should be `UNLOADED` now (which we're now renaming `INACTIVE`)) some 5 seconds before removing the `iframe` from the DOM on close.